From 2fe80030c7e38705323faea91ba714464244e25e Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Wed, 13 Mar 2019 06:44:15 +0530
Subject: [PATCH 037/386] Revert "soc: octeontx2: Get PF macaddress from
 firmware"

This reverts commit 73540f1689c61bd83704857568ab436421fb122a.

Change-Id: Icf9c48043d03d782a22fe846722d3631f4f78e9d
Signed-off-by: Linu Cherian <lcherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/5666
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/af/cgx.c   | 29 -------------------
 .../net/ethernet/marvell/octeontx2/af/cgx.h   |  1 -
 .../ethernet/marvell/octeontx2/af/rvu_cgx.c   |  6 ----
 3 files changed, 36 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 0da0437b434e..18a79a58be6f 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -72,7 +72,6 @@ static char *cgx_lmactype_string[LMAC_MODE_MAX];
 
 /* CGX PHY management internal APIs */
 static int cgx_fwi_link_change(struct cgx *cgx, int lmac_id, bool en);
-static int cgx_fwi_get_macaddr(struct cgx *cgx, int lmac_id, u8 *dst);
 
 /* Supported devices */
 static const struct pci_device_id cgx_id_table[] = {
@@ -198,18 +197,6 @@ u64 cgx_lmac_addr_get(u8 cgx_id, u8 lmac_id)
 }
 EXPORT_SYMBOL(cgx_lmac_addr_get);
 
-/* Gets mac address from firmware for each cgx mapped PF */
-int cgx_get_pfmacaddr(void *cgxd, u8 lmac_id, u8 *macdst)
-{
-	struct cgx *cgx = cgxd;
-
-	if (!cgx || lmac_id >= cgx->lmac_count)
-		return -ENODEV;
-
-	return cgx_fwi_get_macaddr(cgx, lmac_id, macdst);
-}
-EXPORT_SYMBOL(cgx_get_pfmacaddr);
-
 int cgx_set_pkind(void *cgxd, u8 lmac_id, int pkind)
 {
 	struct cgx *cgx = cgxd;
@@ -845,22 +832,6 @@ int cgx_lmac_evh_unregister(void *cgxd, int lmac_id)
 }
 EXPORT_SYMBOL(cgx_lmac_evh_unregister);
 
-static int cgx_fwi_get_macaddr(struct cgx *cgx, int lmac_id, u8 *dst)
-{
-	u64 req = 0, resp, mac;
-	int err;
-
-	req = FIELD_SET(CMDREG_ID, CGX_CMD_GET_MAC_ADDR, req);
-
-	err = cgx_fwi_cmd_generic(req, &resp, cgx, lmac_id);
-	if (!err) {
-		mac = FIELD_GET(RESP_MAC_ADDR, resp);
-		memcpy(dst, (u8 *)&mac, 6);
-	}
-
-	return err;
-}
-
 static int cgx_fwi_link_change(struct cgx *cgx, int lmac_id, bool enable)
 {
 	u64 req = 0;
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.h b/drivers/net/ethernet/marvell/octeontx2/af/cgx.h
index 473b1890ab5c..ef7367df14fb 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.h
@@ -127,7 +127,6 @@ int cgx_lmac_rx_tx_enable(void *cgxd, int lmac_id, bool enable);
 int cgx_lmac_tx_enable(void *cgxd, int lmac_id, bool enable);
 int cgx_lmac_addr_set(u8 cgx_id, u8 lmac_id, u8 *mac_addr);
 u64 cgx_lmac_addr_get(u8 cgx_id, u8 lmac_id);
-int cgx_get_pfmacaddr(void *cgxd, u8 lmac_id, u8 *macdst);
 void cgx_lmac_promisc_config(int cgx_id, int lmac_id, bool enable);
 void cgx_lmac_ptp_config(void *cgxd, int lmac_id, bool enable);
 void cgx_lmac_enadis_rx_pause_fwding(void *cgxd, int lmac_id, bool enable);
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cgx.c
index 7f5d7e8fba8c..d3ae1b3266af 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cgx.c
@@ -100,12 +100,6 @@ static int rvu_map_cgx_lmac_pf(struct rvu *rvu)
 		for (lmac = 0; lmac < lmac_cnt; lmac++, pf++) {
 			rvu->pf2cgxlmac_map[pf] = cgxlmac_id_to_bmap(cgx, lmac);
 			rvu->cgxlmac2pf_map[CGX_OFFSET(cgx) + lmac] = 1 << pf;
-			/* Get mac address for the CGX mapped PF */
-			if (cgx_get_pfmacaddr(rvu_cgx_pdata(cgx, rvu), lmac,
-					  rvu->pf[pf].mac_addr))
-				dev_dbg(rvu->dev,
-					"pf %d: Failed to get MAC from firmware\n",
-					pf);
 			free_pkind = rvu_alloc_rsrc(&pkind->rsrc);
 			pkind->pfchan_map[free_pkind] = ((pf) & 0x3F) << 16;
 			rvu->cgx_mapped_pfs++;
-- 
2.17.1

